nimble/host/src/ble_l2cap_sig.c

```diff
@@ -869,6 +1261,170 @@ ble_l2cap_sig_coc_connect(uint16_t conn_handle, uint16_t psm, uint16_t mtu,
     return rc;
 }
 
+int
+ble_l2cap_sig_coc_connect_multiple(uint16_t conn_handle,
+                                   uint16_t psm, uint16_t mtu,
+                                   uint8_t num, struct os_mbuf *sdu_rx[num],
+                                   ble_l2cap_event_fn *cb, void *cb_arg)
+{
+#if MYNEWT_VAL(BLE_VERSION) >= 52
+    struct ble_hs_conn *conn;
+    struct ble_l2cap_sig_proc *proc;
+    struct ble_l2cap_chan *chan = NULL;
+    struct os_mbuf *txom;
+    struct ble_l2cap_sig_credit_base_connect_req *req;
+    int rc;
+    int i;
+    int j;
+
+    if (!sdu_rx || !cb) {
+        return BLE_HS_EINVAL;
+    }
+
+    ble_hs_lock();
+    conn = ble_hs_conn_find(conn_handle);
+
+    if (!conn) {
+        ble_hs_unlock();
+        return BLE_HS_ENOTCONN;
+    }
+
+    proc = ble_l2cap_sig_proc_alloc();
+        if (!proc) {
+            ble_l2cap_chan_free(conn, chan);
+            ble_hs_unlock();
+            return BLE_HS_ENOMEM;
+        }
+
+    for (i = 0; i < num; i++) {
+        chan = ble_l2cap_coc_chan_alloc(conn, psm, mtu, sdu_rx[i], cb, cb_arg);
+        if (!chan) {
+            for (j = 0; j < i; j++) {
+                chan[j].cb = NULL;
```

Comment:
Can you do this in ble_l2cap_chan_free?

Reply:
this is done for purpose to make sure host will not send "Disconnected callback" to the app.
I will put a comment here.